### PR TITLE
fix: visible groups

### PR DESCRIPTION
**Pull request by thenanyu**
alternative solution to https://github.com/outline/outline/issues/2616

Groups and their members are not secret information generally, so we should try to take a more permissive and less complex approach to begin with.

This PR simply allows the entire team to see every group.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "build:webpack": "webpack --config webpack.config.prod.js",
     "build": "yarn clean && yarn build:webpack && yarn build:i18n && yarn build:server",
     "start": "node ./build/server/index.js",
-    "dev": "yarn concurrently -n api,collaboration  -c \"blue,magenta\" \"node --inspect=0.0.0.0 build/server/index.js --services=collaboration,websockets,admin,web,worker\"",
+    "dev": "NODE_ENV=development yarn concurrently -n api,collaboration  -c \"blue,magenta\" \"node --inspect=0.0.0.0 build/server/index.js --services=collaboration,websockets,admin,web,worker\"",
```
Comment (thenanyu) [inline package.json:13]: This was necessary to get email login working locally to test group behavior; I think it's worth leaving it in this part of the script, as opposed to expecting developers to add it to their `.env` files

File: server/mailer.js [hunk 2]
```diff
@@ -204,12 +204,18 @@ export class Mailer {
   };
 
   signin = async (opts: { to: string, token: string, teamUrl: string }) => {
+    const signInLink = signinEmailText(opts);
+
+    if (process.env.NODE_ENV === "development") {
+      Logger.info(`Sign-In link: ${signInLink}`);
+    }
```
Comment (thenanyu) [inline server/mailer.js:211]: I can go either way with leaving this in, but it's useful when the dev mail service is unavailable or if the developer doesn't have a network connection

File: server/routes/api/documents.test.js [hunk 3]
```diff
@@ -1230,7 +1230,11 @@ describe("#documents.search", () => {
   });
 
   it("should return documents for a specific private collection", async () => {
-    const { user, collection } = await seed();
+    const user = await buildUser();
+    const collection = await buildCollection({
+      teamId: user.teamId,
+    });
+
```
Comment (thenanyu) [inline server/routes/api/documents.test.js:1237]: tried to move towards model-specific builders for tests to possibly solve deadlock problems; leaving them in as it's a general improvement

Comment (tommoor) [reply server/routes/api/documents.test.js:1237]: Yea, I've been trying to remove `seed` gradually to make things faster too

**Review by tommoor — APPROVED** [review]

**Comment by tommoor** [comment]
Are you wanting to change anything else here or go to merge?

**Comment by thenanyu** [comment]
I'll merge it. Did it on the plane but internet failed

**Pull request merged**